Snip/Players/Spotify.cs

```diff
@@ -261,61 +261,94 @@ private void ContactSpotifyLocalServerTimer_Elapsed(object sender, ElapsedEventA
                         this.ResetSnipSinceSpotifyIsNotPlaying();
                     }
                     else
-                    {
-                        string fullTrackId = jsonSummary.track.track_resource.uri.ToString();
-                        string trackId = fullTrackId.Substring(fullTrackId.LastIndexOf(':') + 1); // + 1 to not include :
-
-                        // Only update if the title has changed or the user updates how the output format should look
-                        if (trackId != this.LastTitle || Globals.RewriteUpdatedOutputFormat)
+                    {                        
+                        if (!LocalTrackUpdateText(jsonSummary))
                         {
-                            Globals.RewriteUpdatedOutputFormat = false;
+                            SpotifyTrackUpdateText(jsonSummary);
+                        }
+                    }
+                }
+            }
+        }
 
-                            string json = string.Empty;
+        private bool LocalTrackUpdateText(dynamic jsonSummary)
+        {
+            // Local tracks have limited information available, and it can't be
+            // fetched thru the Spotify API.
+            if (jsonSummary.track.track_type.ToString() == "local")
+            {
+                var trackId = jsonSummary.track.track_resource.uri;
 
-                            if (Globals.CacheSpotifyMetadata)
-                            {
-                                json = this.ReadCachedJson(trackId);
-                            }
-                            else
-                            {
-                                json = this.DownloadJson(
-                                    string.Format(
-                                        CultureInfo.InvariantCulture,
-                                        "https://api.spotify.com/v1/tracks/{0}",
-                                        trackId),
-                                    SpotifyAddressContactType.API);
-                            }
+                if (trackId != this.LastTitle || Globals.RewriteUpdatedOutputFormat)
+                {
+                    Globals.RewriteUpdatedOutputFormat = false;
 
-                            jsonSummary = SimpleJson.DeserializeObject(json);
+                    var name = jsonSummary.track.track_resource.name;
+                    var artist = jsonSummary.track.artist_resource.name;
```

Comment:
Does this work with tracks with multiple artists, or does the API only give you the first one anyway?

Reply:
The summary does indeed only list the first artist in the `track_resource`. However, local tracks can be tied to spotify tracks, so the local should be a fallback rather than a first attempt. I'll make that change.

![image](https://user-images.githubusercontent.com/3794983/33190734-fe58eb44-d064-11e7-98f9-74d9e2786d0c.png)